### ISSUE TITLE
GRPO OOM: Qwen 72B (6xH100 Training, 4xH100 External vLLM) - Master GPU Low Load & Crash

### ISSUE THREAD

**Issue by MSS444** [modelscope/ms-swift#4440]
**description**
i am doing GRPO training for Qwen 72B model config and logs are attached i have 10 h100 80gb gpus, i am using external vllm server, 4 gpus for vllm 6 for training, i get OOM in training, the issue is that the master GPU doesn't get any memory loaded on it, can you please help, I am using the latest version of ms-swift (by cloning the repo) if any other info is required let me know, thank you. 

![Image](https://github.com/user-attachments/assets/109cf72b-8afd-4fdf-9096-9bc874d696bd)

**config:**
training_params:

  **Base parameters**
  rlhf_type: "grpo"
  output_dir: "outputs_non_reasoning"
  logging_dir: "outputs_non_reasoning/runs"
  train_type: "lora"    
  lora_rank: 32  
  torch_dtype: "bfloat16"
  async_generate: true 
  use_vllm: true
  vllm_mode: server
  vllm_server_host: "0.0.0.0"
  vllm_server_port: 8000
  vllm_server_timeout: 600
  CUDA_VISIBLE_DEVICES_MS_SWIFT: 0,7,5,6,8,9
  NPROC_PER_NODE: 6

  **vLLM parameters**
  vllm_max_model_len: 1024 #prompt + generation
  CUDA_VISIBLE_DEVICES_VLLM: 1,4,2,3
  tensor_parallel_size: 4
  model: "Qwen/Qwen2.5-72B"
  model_type: "qwen2_5"
  gpu_memory_utilization: 0.9

  **Optimization parameters**
  learning_rate: 0.00001  # 1e-5 ->> 1e-6
  warmup_ratio: 0.05
  
  **Training hyperparameters**
  num_train_epochs: 10
  #max_steps: 1
  per_device_train_batch_size: 1
  per_device_eval_batch_size: 1
  gradient_accumulation_steps: 1
  #eval_steps: null # -> commented this out
  split_dataset_ratio: 0.10 # -> added this, 30% data for validation i.e., 4 examples
  save_steps: 5
  eval_strategy: "epoch"  #-> changed from no to epoch
  save_total_limit: 2
  max_length: 512 # max prompt length beyond which we truncate
  max_completion_length: 512 #this much generation for each prompt in GRPO trainer
  gradient_checkpointing: true

  **Logging parameters**
  logging_steps: 1
  log_completions: true
  report_to: tensorboard
  
  **Data loading parameters**
  dataloader_num_workers: 6

  **Generation parameters**
  num_generations: 2 #total number of completions generated, divide by no of gpus to get per gpu completions
  temperature: 1

  **Reward functions**
  external_plugins: "/ps/training/rl_ft/ms_swift/plugin_functions.py"
  reward_funcs: ["format_reward", "brs"]
  reward_weights: [0.3 , 0.7]
  
  **DeepSpeed configuration**
  deepspeed: "zero3_offload"

**logs**
[logs.txt](https://github.com/user-attachments/files/20548111/logs.txt)

**Comment by hjh0119** [comment]
For OOM in the gather phase, you can use the parameter --move_model_batches to gather in batches. See details in [this pr](https://github.com/modelscope/ms-swift/pull/4453).

**Comment by MSS444** [comment]
thank you.
